### PR TITLE
MHV-43924 SM - [Added test case for expand all messages and focus them by pressing nav]

### PR DESCRIPTION
**Pull request by haniasma**
## Summary

Added new automation test case for Expand all messages and verify the focus before and after pressing enter on each message.

## Screenshots

image

### CODE REVIEW

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]
Let's take it as a regular practice to keep PR descriptions updated to reflect the scope of work captured in PR. Also, it will be helpful to keep PR title in the following format:
`MHV-[TICKET_NUMMBER] SM - [PR Title]`

File: src/applications/mhv-secure-messaging/main/tests/e2e/keyboard-nav-test/secure-messaging-keyboard-nav-to-message-details.cypress.spec.js [hunk 1]
```diff
@@ -27,6 +27,45 @@ describe('Navigate to Message Details ', () => {
...
+    cy.get(Locators.BUTTONS.BUTTON_MOVE).should('have.focus');
+    cy.realPress('Tab');
+    cy.get(Locators.BUTTONS.BUTTON_TRASH)
+      .should('be.visible')
+      .then(() => {
+        cy.get(Locators.BUTTONS.BUTTON_TRASH).should('have.focus');
+      });
+    for (let i = 0; i < 9; i++) {
+      cy.realPress('Tab');
+    }
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/main/tests/e2e/keyboard-nav-test/secure-messaging-keyboard-nav-to-message-details.cypress.spec.js:66]: I think I understand the goal this loop is trying to reach, which is tab to "Expand All +" button. I am concerned that an arbitrary number of iterations here may not be reliable. is there a way to do a `while` loop with a condition when a specific button reaches focus?

File: src/applications/mhv-secure-messaging/main/tests/e2e/pages/PatientMessageDetailsPage.js [hunk 2]
```diff
@@ -501,6 +501,33 @@ class PatientMessageDetailsPage {
     );
   };
 
+  verifyClickAndExpendAllMessagesHasFocus = () => {
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/main/tests/e2e/pages/PatientMessageDetailsPage.js:504]:
```suggestion
  verifyClickAndExpandAllMessagesHasFocus = () => {
```

**Review by powellkerry — DISMISSED** [review]

**Review by Asterix22 — APPROVED** [review]

**Review by alex-morgun-1 — APPROVED** [review]
👍🏻

**Review by fazilqa — APPROVED** [review]
all requests resolved. LGTM

**Pull request merged**